src/library/scala/math/Ordering.scala

```diff
@@ -122,6 +122,49 @@ trait Ordering[T] extends Comparator[T] with PartialOrdering[T] with Serializabl
     def compare(x: U, y: U) = outer.compare(f(x), f(y))
   }
 
+  /** Creates an Ordering[T] whose compare function returns the
+    * result of this Ordering's compare function, if it is non-zero,
+    * or else the result of `other`s compare function.
+    *
+    * @example
+    * {{{
+    * case class Pair(a: Int, b: Int)
+    *
+    * val pairOrdering = Ordering.by[Pair, Int](_.a)
+    *                            .orElse(Ordering.by[Pair, Int](_.b))
+    * }}}
+    *
+    * @param other an Ordering to use if this Ordering returns zero
+    */
+  def orElse(other: Ordering[T]): Ordering[T] = (x, y) => {
+    val res1 = outer.compare(x, y)
+    if (res1 != 0) res1 else other.compare(x, y)
+  }
+
+  /** Given f, a function from T into S, creates an Ordering[T] whose compare
+    * function returns the result of this Ordering's compare function,
+    * if it is non-zero, or else a result equivalent to:
+    *
+    * {{{
+    * Ordering[S].compare(f(x), f(y))
+    * }}}
+    *
+    * This function is analogous to passing the result of `Ordering.by(f)`
```

Comment:
Yes, I suppose it is (though slightly cheaper in terms of allocation).

I perhaps should not have borrowed the wording from `by` as much.